### ISSUE TITLE
Some issues with epub to Kindle

### ISSUE THREAD

**Issue by crazybadger** [janeczku/calibre-web#2508]
Firstly, a big thanks for the update to send epub files to Kindle.  It's so much easier to wirelessly send books to my Kindle than to connect it up with a cable to transfer.

I'm having an issue with strange characters appearing, though, and wonder if that was an issue with Calibre-Web sending or Amazon transferring to my Kindle?

I've only tried a few books (all starting in epub format and look fine when viewing on Calibre-Web), but they all have â€™️ inserted where I think there should be an apostrophe, for example, whoâ€™️d instead of who'd.  There are other combinations which I haven't quite worked out yet, like â€ and â€œ and â€̃̃̃ although these appear less often.

I'm running Calibre-Web 0.6.19 in a Docker container in a ubuntu 20.4 VM on a TrueNAS server and accessing through Chrome 103.0.5060.134.

It's quite hard to read around these, so any help would is much appreciated.

Thanks

**Comment by OzzieIsaacs** [comment]
It‘s an encoding problem. Please check if it also occurs if you send the books directly to you kindle (via your email program). There are also some other reports regarding this. I didn’t had time to check the cause and find a solution

**Comment by OzzieIsaacs** [comment]
https://github.com/janeczku/calibre-web/issues/2393#issuecomment-1183213405
have also a look here

**Comment by crazybadger** [comment]
Yes, I just tried sending to the e-mail address using my mail client and I get the same issue, although with some slightly different formatting.   I also found this online, which might help?  https://blog.the-ebook-reader.com/2022/05/06/how-to-fix-formatting-issues-on-epubs-sent-to-kindles.  Is it possible to set “Encode HTML in UTF-8” in Calibre-Web?

**Comment by OzzieIsaacs** [comment]
This is a calibre plugin, calibre-web can’t use it

**Comment by crazybadger** [comment]
I just hoped it might be something that could be built into Calibre-Web.  I reported this as an issue with Amazon too, and they said it was something they were aware of and were working on a fix, but no timescales.  Back to converting epub to azw and then transferring with a cable :-(

**Comment by alex21311** [comment]
Hi. Pleae have a look at this: https://www.mobileread.com/forums/showpost.php?p=4233972&postcount=3

I would propose to check first on existing azw3 format before sending to e-reader. If it exists, just send the epub. If not, convert epub to azw3, delete epub (or rename to .origin_epub) and convert azw3 back to epub. This is a similar solution as suggested by calibre (https://manual.calibre-ebook.com/faq.html#amazon-is-stopping-email-delivery-of-mobi-files), but when converting to azw3 and back you shouldn't loose advanced formating.

**Comment by crazybadger** [comment]
Thank you!  That works perfectly on the few books I've tried.  It would be great if that could be built into the export routine by default?

**Comment by OzzieIsaacs** [comment]
I can confirm the method of @alex21311 is also working for me. The other suggested methods didn't work for me. So I guess it's not a encoding problem alone, there is more behind it. So it could be harder to fix, and if amazon is aware of it, it's maybe waste of time (as they will have their fix ready at the same time I will provide a solution)

**Comment by Izak90** [comment]
Also if you convert from EPUB to EPUB in calibri will do the magic too. You will have two formats: EPUB and ORIGINAL_EPUB

**Comment by bmn001** [comment]
Thank you @alex21311!  Your steps definitely helped.  My books looked fine in Calibre-Web, but would get loaded with crazy characters once sent to my Kindle.  Your process is a good workaround for now.

**Comment by olskar** [comment]
Theres also this tool 

https://kindle-epub-fix.netlify.app/

**Comment by crazybadger** [comment]
Thanks!  It must be possible to build this into Calibre-web as it's just converting to azw3 and back again?  The workaround had been perfect for me, but it's about ten extra clicks (not including checking to see if the conversions are complete) when it used to be a single click of the 'send to ereader' button

**Comment by elp3dr0** [comment]
I've just come up against this problem. Is the recommended workflow still  EPUB ->AZW -> EPUB step by step in calibre-web, or has anyone found a more streamlined/automated/batched approach?

**Comment by ljr95** [comment]
Hi,
Will there be an update for this issue?
Thank you

**Comment by thrust26** [comment]
Thanks for the AZW3 tip! Without the trick, 1 out of 5 eBooks wouldn't send.

**Comment by johnjaylward** [comment]
I'm having an issue where some books show the "Send Epub to e-reader" button, but others don't. Why would some books not show that option?
